src/datasources/accounts/accounts.datasource.spec.ts

```diff
@@ -88,20 +91,126 @@ describe('AccountsDatasource tests', () => {
       );
     });
 
-    it('throws when an account with the same address already exists', async () => {
+    it('creates an account successfully if the clientIp is not a valid IP', async () => {
       const address = getAddress(faker.finance.ethereumAddress());
-      await target.createAccount(address);
 
-      await expect(target.createAccount(address)).rejects.toThrow(
-        'Error creating account.',
+      const result = await target.createAccount({
+        address,
+        clientIp: faker.string.sample(),
+      });
+
+      expect(result).toStrictEqual({
+        id: expect.any(Number),
+        group_id: null,
+        address,
+        created_at: expect.any(Date),
+        updated_at: expect.any(Date),
+      });
+
+      // check the account is stored in the cache
+      const cacheDir = new CacheDir(`account_${address}`, '');
+      const cacheContent = await fakeCacheService.get(cacheDir);
+      expect(JSON.parse(cacheContent as string)).toStrictEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            id: expect.any(Number),
+            group_id: null,
+            address,
+          }),
+        ]),
       );
     });
+
+    it('should fail if the IP hits the rate limit', async () => {
+      const clientIp = faker.internet.ipv4();
+      const accountCreationRateLimitCalls = faker.number.int({
+        min: 2,
+        max: 5,
+      });
+      mockConfigurationService.getOrThrow.mockImplementation((key) => {
+        if (key === 'expirationTimeInSeconds.default')
+          return faker.number.int();
+        if (key === 'accounts.creationRateLimitCalls')
+          return accountCreationRateLimitCalls;
+        if (key === 'accounts.creationRateLimitPeriodSeconds')
+          return faker.number.int({ min: 10 });
+      });
+      target = new AccountsDatasource(
+        fakeCacheService,
+        sql,
+        new CachedQueryResolver(mockLoggingService, fakeCacheService),
+        mockLoggingService,
+        mockConfigurationService,
+      );
+
+      for (let i = 0; i < accountCreationRateLimitCalls; i++) {
+        await target.createAccount({
+          address: getAddress(faker.finance.ethereumAddress()),
+          clientIp,
+        });
+      }
+
+      await expect(
+        target.createAccount({
+          address: getAddress(faker.finance.ethereumAddress()),
+          clientIp,
+        }),
+      ).rejects.toThrow('Rate limit reached');
+
+      expect(mockLoggingService.warn).toHaveBeenCalledTimes(1);
```

Comment:
I'm impartial regarding this but I don't think it's imperative we check that logging occurred. (I won't comment regarding this below.)

Reply:
Removed in [b2e5161](https://github.com/safe-global/safe-client-gateway/pull/1800/commits/b2e51613dcbc65252b8e6f2865027a72adc25644)